Update CS0524 documentation to reflect C# 8.0 changes allowing nested types in interfaces (#47669)
dotnet/docs: docs/csharp/misc/cs0524.md

'type' : interfaces cannot declare types  
  
Prior to C# 8.0, an [interface](../language-reference/keywords/interface.md) could not contain user-defined types. Starting with C# 8.0, interfaces can declare [nested types](../programming-guide/classes-and-structs/nested-types.md) as part of the default interface members feature.

In modern C# (C# 8.0 and later), this error is no longer generated for nested types in interfaces. The compiler now allows interfaces to contain nested classes, structs, interfaces, enums, and delegates.

## Historical example

> The following sample would have generated CS0524 in C# versions prior to 8.0, but is now valid code:  

Reviewer: ```suggestion
The following sample generates CS0524 in C# versions prior to 8.0, but is valid in modern C# versions:  
```
